flake.nix

```diff
@@ -94,157 +92,166 @@
           '';
         };
 
+        # Custom packages and aliases for building hosts
+        # Accessible through 'nix build', 'nix run', etc
         packages = with flake.nixosConfigurations; {
-          "homestakeros" = homestakeros.config.system.build.kexecTree;
+          "homestaker" = homestaker.config.system.build.kexecTree;
         };
       };
       flake =
         let
           inherit (self) outputs;
 
-          homestakeros_options = with nixpkgs.lib; {
-            localization = {
-              hostname = mkOption {
-                type = types.str;
-                default = "homestaker";
-                description = "The name of the machine";
-              };
-              timezone = mkOption {
-                type = types.str;
-                default = "Europe/Helsinki";
-                description = "The time zone used when displaying times and dates";
+          homestakerOptions = with nixpkgs.lib;
+            let
+              nospace = str: filter (c: c == " ") (stringToCharacters str) == [ ];
+            in
+            {
+              localization = {
+                hostname = mkOption {
+                  type = types.strMatching "^$|^[[:alnum:]]([[:alnum:]_-]{0,61}[[:alnum:]])?$";
+                  default = "homestaker";
+                  description = "The name of the machine.";
+                };
+                timezone = mkOption {
+                  type = types.nullOr (types.addCheck types.str nospace);
+                  default = null;
+                  description = "The time zone used when displaying times and dates.";
+                  example = "America/New_York";
+                };
               };
-            };
-
-            mounts = mkOption {
-              type = types.attrsOf types.attrs;
-              default = { };
-              description = "Systemd mounts configuration";
-            };
 
-            ssh = {
-              privateKeyPath = mkOption {
-                type = types.path;
-                default = "/var/mnt/secrets/ssh/id_ed25519";
-                description = "Path, where SSH host keys are automatically generated";
+              mounts = mkOption {
+                type = types.attrsOf types.attrs;
+                default = { };
+                description = "Systemd mounts configuration.";
               };
-            };
 
-            user = {
-              authorizedKeys = mkOption {
-                type = types.listOf types.singleLineStr;
-                default = [ ];
-                description = "A list of public keys that should be added to the user’s authorized keys";
+              ssh = {
+                privateKeyFile = mkOption {
+                  type = types.nullOr types.path;
+                  default = null;
+                  description = "Path to automatically generated Ed25519 SSH host key.";
```

Comment:
This description should be reconsidered: the Ed25519 SSH host key will be automatically created if one does not exist, but this option is here to pin it to some pre-defined key.

Reply:
Also, can you look into can this use markdown?

Reply:
It has to be a single or multi-line string. Basically, it could store markdown for frontend.

Reply:
Add it as Markdown, so we can try it later on.